api/src/main/java/marquez/service/RunService.java

```diff
@@ -66,36 +66,36 @@ public void markRunAs(
     if (transitionedAt == null) {
       transitionedAt = Instant.now();
     }
-    ExtendedRunRow runRow = findByRow(runId.getValue()).get();
-    runStateDao.updateRunState(runId.getValue(), runState, transitionedAt);
+    ExtendedRunRow runRow = findRunByUuidAsRow(runId.getValue()).get();
+    runStateDao.updateRunStateFor(runId.getValue(), runState, transitionedAt);
 
     if (runState.isDone()) {
-      JobVersionBag jobVersionBag =
-          jobVersionDao.createJobVersionOnComplete(
-              transitionedAt,
-              runRow.getUuid(),
+      BagOfJobVersionInfo bagOfJobVersionInfo =
+          jobVersionDao.upsertJobVersionOnRunTransition(
               runRow.getNamespaceName(),
               runRow.getJobName(),
-              runState);
+              runRow.getUuid(),
+              runState,
+              transitionedAt);
 
       if (runState == COMPLETED) {
```

Comment:
Should this conditional match the new JobVersion logic? I.e., should this happen when `run.isDone()`?

Reply:
I _think_ it should, but, at Datakin, we rely on the `COMPLETED` state transition to build the lineage graph. So, though I agree the logic should be the same, without proper testing around the state transition, I'd like to hold off on making the change in this PR. I'd recommend a follow up PR with proper testing.

Reply:
Sounds good. Will you leave a TODO in the code?

Reply:
Yep 👍